bluetooth/src/androidLibMain/kotlin/device/DeviceConnectionManager.kt

```diff
@@ -195,12 +203,16 @@ internal actual class DeviceConnectionManager(
                 gatt.await().setCharacteristicNotification(action.characteristic.wrapper, action.enable).also {
                     info(TAG, "setCharacteristicNotification result: $it")
                 }
-                // Enable/Disable remote notifications
-                val value = if (action.enable) BluetoothGattDescriptor.ENABLE_NOTIFICATION_VALUE
+
+                info { " * wrapper property: ${action.characteristic.wrapper.properties}" }
+                val value = if (action.enable)
+                    if (action.characteristic.wrapper.hasProperty(PROPERTY_INDICATE))
+                        BluetoothGattDescriptor.ENABLE_INDICATION_VALUE
+                    else BluetoothGattDescriptor.ENABLE_NOTIFICATION_VALUE
                 else BluetoothGattDescriptor.DISABLE_NOTIFICATION_VALUE
```

Comment:
`when` would be more readable

Reply:
Then we also can check `hasProperty(PROPERTY_NOTIFY)` to avoid calls for non notifiable characteristics